demophin.py

```diff
@@ -91,8 +109,17 @@ def parse_sentence(grm, sent, n=None):
         return None
     # update cmdargs as necessary
     opts = dict(ace_options)
+    # use optional ACE args
+    if grm.get('aceopts'):
+        opts['cmdargs'] += grm.get('aceopts')
+    # update n properly
     if n is not None:
-        opts['cmdargs'] = opts['cmdargs'] + ['-n', str(n)]
+        for i in range(len(opts['cmdargs'])):
+            if opts['cmdargs'][i].startswith('-n '):
+                opts['cmdargs'][i] = '-n ' + str(n)
+    #  uniqify
+    opts['cmdargs'] = list(set(opts['cmdargs']))
```

Comment:
I accidentally commented on the wrong commit for this line. Basically, don't call `set` on this. First of all, arguments and values should be separate (`["-n", "10"]`, not `["-n 10"]`), and because of this calling `set` will be destructive because it removes the order. Secondly, joining the arguments (e.g. `-n 10`) means that even if there were more than one, set wouldn't catch it (`"-n 10"` and `"-n 5"` are different strings). Because ACE uses the last argument value, we would randomly get an "-n" argument (and also making them unique isn't necessary).